### DISCUSSION TITLE
Cursed C# - Doing shenanigans in C#

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#22] [General]
# Cursed C# - Doing shenanigans in C#

In this short blog post I want to show you two silly things so that you can apply right now! Both of them equally silly, but that is not the point (is it ever?).
We will see how to await an integer or TimeSpan and how to foreach through an integer. All of this thanks to the magic of extensions methods.

https://steven-giesel.com/blogPost/5360d1c3-89f6-4a08-9ee3-6ddbe1b44236

**Comment by kcragin** [comment]
This is downright diabolical, and it gets worse. If you change the iterator so that `Current` returns a `string`, you could create an iterator such that `foreach (var v in 4)` would iterate out to "one", "two", "three", "four".

I will now go turn myself in to the authorities.

**Reply by linkdotnet** [reply]
Haha yes, I like the spirit. To be honest I am not sure if I can find any valid use case for that.  :D
